config/config-mp/src/main/java/io/helidon/config/mp/MpConfigImpl.java

```diff
@@ -126,70 +129,88 @@ public <T> T getValue(String propertyName, Class<T> propertyType) {
     @SuppressWarnings("unchecked")
     @Override
     public <T> Optional<T> getOptionalValue(String propertyName, Class<T> propertyType) {
+        String profiledPropertyName = null;
```

Comment:
This statement does not make sense here - the next if is checking for profile being null. So just move the content of the if statement after the if checking if profile is null and remove the check.

Reply:
Done